master/buildbot/test/fake/botmaster.py

```diff
@@ -24,22 +25,16 @@ class FakeBotMaster(service.AsyncMultiService):
     def __init__(self):
         super().__init__()
         self.setName("fake-botmaster")
-        self.locks = {}
+        self.lock_tracker = BotLockTracker()
         self.builders = {}  # dictionary mapping worker names to builders
         self.buildsStartedForWorkers = []
         self.delayShutdown = False
 
-    def getLockByID(self, lockid):
-        if lockid not in self.locks:
-            self.locks[lockid] = lockid.lockClass(lockid)
-        # if the master.cfg file has changed maxCount= on the lock, the next
-        # time a build is started, they'll get a new RealLock instance. Note
-        # that this requires that MasterLock and WorkerLock (marker) instances
-        # be hashable and that they should compare properly.
-        return self.locks[lockid]
-
-    def getLockFromLockAccess(self, access):
-        return self.getLockByID(access.lockid)
+    def getLockByID(self, lockid, config_version):
+        return self.lock_tracker.getLockByID(lockid, config_version)
```

Comment:
BotMaster is not a SharedService, it is a BuildbotService.
SharedService are used to manage singletons. The only instance of SharedService now are HttpClientService and its fake.

To use it, you just have to change this line to:
```suggestion
        return SharedService.getService(lockid.lockClass, self, lockid)
```

then add SharedService parent class to RealWorkerLock and RealMasterLock

Reply:
Okay, I completely misunderstood how exactly SharedService works. It's much clearer now. I've rewritten the code to use `SharedService` now.